### ISSUE TITLE
[Feature Request] Implementation of common Office short-cuts, mainly "F4"

### ISSUE THREAD

**Issue by rustysys-dev** [ONLYOFFICE/DesktopEditors#79]
**Do you want to request a *feature* or report a *bug*?**
Feature

**What is the current behavior?**
No support for many common short cuts.

**What is the expected behavior?**
As in WPS Office or MS Office, I would like to see many of the same short-cuts for the editors.  Namely, the implementation of the "F4" ( repeat last action ) short-cut.  This is paramount for formatting as it drastically reduces the amount of time required.

**Which versions of DesktopEditors, and which OS are affected by this issue? Did this work in previous versions of DesktopEditors?**
I have installed only the newest version, but I would assume that if this version doesn't have that feature then neither do the others... correct me if I am wrong.

**Comment by vatterspun** [comment]
I'm surprised this hasn't come up but I don't see anything in the issues listing.

**Configuration**

Because of the possible overlap with a given global hotkey, the keys are going to need to be something you can change.  So it's not just (for example) CTRL+B to make text bold, you also need to modify that functionality via a separate menu.

**Priority keys** 

These should probably surround basic operations (like repeat action mentioned by @procyclinsur as well as formatting and outlining functions, since those are where word processors often really shine versus other types of editors.  Things like headline, bold, italics, add bullets, numbering, increase/decrease indent are ideal.  Making it so users won't have to take their fingers off the keyboard just means better productivity.

**Considerations**

I'd encourage taking some time on this one for a few reasons:

1. This is a cross-platform program and whatever key combinations are enabled, they have to consider Windows, Mac, and Linux global shortcuts, which might interfere.
2. At the same time there are many shortcuts which are common between programs, such as CTRL+Tab moving between window tabs in a given program.
3. Key combos should generally make sense to the program and other key combinations.  For example, Microsoft Word uses CTRL+D for fonts because CTRL+F is find.  In Firefox CTRL+D is add bookmark because CTRL+B is already opening the bookmarks menu.  
4. You might run out of hot keys.  For example, Word 2010 doesn't have a hotkey for creating bullets (you have to manually add it) but LibreOffice has for years CTRL+F10.

In other words, a rush to add without considering how they might be used and where there's going to be overlap, and what the more popular functions are going to be.  

Certainly ONLYOFFICE is in early development so some hotkeys could change over time as things progress, but I know this topic is one area where thinking ahead could save time later.

**Comment by jlemangarin** [comment]
Hello @ShockwaveNN !

Any news or ETA about this feature request ? One of our customers are asking for it and we would like to know if this is planned in your roadmap.

Thanks in advance !

Best,

**Comment by ShockwaveNN** [comment]
@jlemangarin Hi, seems this issue is pretty old and I think we were not careful enough in 2017 to correctly handle all issues
Also for me seems there is a lot of features asked in one single issue, so could you describe that exactly do you mean?
F4 for repeat last action or general ability to change shortcuts?

**Comment by ShockwaveNN** [comment]
Oops, closed by misclick

**Comment by jlemangarin** [comment]
Hello @ShockwaveNN,

Yes absolutly, I'm talking about the F4 key to repeat last action.

Thank you ! :)

**Comment by Rita-Bubnova** [comment]
The MS Office documentation says "F4 alone:   repeats the last command or action, if possible.".
It's hard for me to understand when pressing F4 repeats the last command or when action, if possible.

@jlemangarin сould you describe in more detail what you need to do (for example in MS Office)?

**Comment by jlemangarin** [comment]
Hello @Rita-Bubnova,

After some discussions in our forums (In French but cf. https://community.ceo-vision.com/topic/466/onlyoffice-touche-de-raccourci-f4/3) here are some use cases examples : 
- In Excel, after typing smth in a cell, doing F4 in another cell will rewrite the same text
- Inserting an image or a table and doing F4 will redo this action (inserting a second table..)

The main purpose of the F4 key is to redo the last action that we made (If we deleted something, then this key will be a suppr key etc...)

I hope this is clearer now, don't hesitate to ask me for more details.

Thanks in advance !

**Comment by lixaotec** [comment]
+1. 

F4 shortcut is needed, mainly for inserting new rows

**Comment by Rita-Bubnova** [comment]
Sorry for late response.
We  discuss this issue with our team in issue 40557 in our private issue tracker.

**Comment by HectorB-2020** [comment]
Oh it's been so long.
I'm badly missing this. :cry: Are there chances you are going to implement this nice feature?
Please. :cat:

**Comment by ShockwaveNN** [comment]
@HectorB-2020 no, sorry there is no progress. 

Seems `repeat` is not an easy task to implement, it's not like we need just add a shortcut for existing methods
But I'll remind our devs

Also oving this issue to DocumentServer repo, since it's our base product and problem actual there too
